client/layout/guided-tours/component.jsx

```diff
@@ -37,17 +37,14 @@ class GuidedToursComponent extends Component {
 
 	UNSAFE_componentWillReceiveProps( nextProps ) {
 		if ( nextProps.requestedTour === 'reset' && this.props.requestedTour !== 'reset' ) {
-			this.props.resetGuidedToursHistory();
+			this.props.dispatch( resetGuidedToursHistory() );
```

Comment:
I'm having trouble figuring out how the component gets the `dispatch` prop populated. I get that it's coming from context, but I don't see it being defined anywhere, only being forwarded.

In any case, I can just directly ask what I was looking for: is there any guarantee that this `dispatch` is a memoised value that won't change on every render?

Reply:
The `dispatch` prop is coming from `connect`. If `connect` doesn't have the second `mapDispatchToProps` parameter, it will [pass `dispatch` by default](https://react-redux.js.org/api/connect#mapdispatchtoprops-object-dispatch-ownprops-object).

It doesn't change on every render: it's constant for the whole lifetime of the Redux store. Similarly, when binding action creators in `mapDispatchToProps`, they also don't change all the time.

Reply:
Perfect, thank you for clarifying that, @jsnajdr !